Certificate pinning for server trust verification

eli-earnin opened the pull request:
## Summary
Pinning the following URLs against the bundled certificates
- cdn.optimizely.com
- logx.optimizely.com
- api.optimizely.com

To verify endpoint server trust.

## Test plan
Using Charles on the network traffic, create a self sign certificate and replace the certificate from coming from the server with Charles' .
The connection should fail.

## Issues
https://earnin.atlassian.net/browse/MOB-826

EliHini committed:
Certificate pinning for server trust verification

Pinning the following URLs against the bundled certificates

- cdn.optimizely.com
- logx.optimizely.com
- api.optimizely.com

mc-earnin commented:
Hi @eli-earnin 
Pinning the domain(leaf) certificate is generally not a good idea because when the cert expires , HTTPS calls to that domain will fail and this will require a force upgrade to all mobile clients with the new cert. This is even harder for an SDK provider since they don't really control mobile app upgrades for apps using their SDK. 

There are 3 schemes that are implemented in practice -
1) Public Key pinning. 
Since public keys are typically long-lived, it is preferential to check the validity of the public key instead of the entire cert. Most open-source Networking libs like AlamoFire include this option
2) Validate the Root/CA cert only 
Since Root/CA certs are long lasting there is little risk of them expiring . Also the mobile OS will ensure that Root Certs are properly upgraded on the device so that they are always active
3) Keep 2 certs for each domain - one current and one that will be used after the current one expires. The app then checks the current cert first before checking the future one. This works when we have control over the API server and can co-ordinate deployment on both client and server

@jaz-ah @gowthamyemme 
one of our certs for a pinned domain  (www.mogl.com) will expire in November. 
<img width="852" alt="Screen Shot 2019-07-04 at 5 19 51 AM" src="https://user-images.githubusercontent.com/51131443/60668181-2e091500-9e20-11e9-9fbd-5bd39913bc2c.png">

We should move towards public key pinning. 
Specifically this line should be changed to `pinKeys`
https://github.com/activehours/ios/blob/b1ae91265ae0b4d3b25284bace3527576cf7b2aa/AHPlatform/SwaggerClient/Classes/Swaggers/General/AlamofireImplementations.swift#L132

EliHini committed:
Updated certificate validation logic

using pinned cert as anchor cert for verification in `SecTrustEvaluate`. This piece was mission in initial push.

Removed redundant code (** no longer need to rebuild cert from arrays of certs)
